pmd-core/src/main/java/net/sourceforge/pmd/AbstractPropertyDescriptorFactory.java

```diff
@@ -28,30 +30,42 @@
  */
 public abstract class AbstractPropertyDescriptorFactory<T> implements PropertyDescriptorFactory<T> {
 
-    protected static final Map<PropertyDescriptorField, Boolean> CORE_FIELD_TYPES_BY_KEY
-        = CollectionUtil.mapFrom(new PropertyDescriptorField[] {NAME, DESCRIPTION, DEFAULT_VALUE, DELIMITER},
-                                 new Boolean[] {true, true, true, false});
+
+
+
+    static {
+        System.err.println(CORE_FIELD_TYPES_BY_KEY);
```

Comment:
Could it be a static initialization cycle?  I guess I've not hit issues with this before.  Try this to see what I mean:

```
public class Test1 {
    public static final Test2 TEST2 = new Test2();

    static {
        System.out.println("test1 -> test2: " + Test2.TEST1);
    }
    public static void main(final String[] args) {
        System.out.println("main called");
    }
}
```

```
public class Test2 {
    public static final Test1 TEST1 = new Test1();
    static {
        System.out.println("test2 -> test1: " + Test1.TEST2);
    }
}
```
When I run, I get this output:
```
test2 -> test1: null
test1 -> test2: com.firepond.cmod.Test1@7852e922
main called
```

You'll need to track down the class initialization cycle and find a way to break it.

When I'm designing a non-trivial subsystem, I sometimes like to get a tool to visualize the real dependencies between the different classes.  A picture can make it immediately obvious where I've introduced a connection that breaks the "layering" or intended encapsulation.  I tend to shoot for trees if I can, DAGs if I must, and spend a lot of time trying to eliminate cycles. 
 It's also useful to check the implementation meets my design goals.  Perhaps helpful here to understand where the cycle exists.